pymemcache/client/base.py

```diff
@@ -1249,6 +1280,10 @@ def quit(self):
             finally:
                 self.client_pool.destroy(client)
 
+    def shutdown(self, graceful=False):
+        with self.client_pool.get_and_release(destroy_on_fail=True) as client:
+            client.shutdown(graceful)
```

Comment:
Should the client be destroyed on a successful shutdown?

```suggestion
            client.shutdown(graceful)
            self.client_pool.destroy(client)
```

Reply:
I'm not sure; I guess it depends on the desired interface contract. The changeset as it is now proposes that this API be a relatively thin wrapper for server shutdown, i.e. changing the state of the server without changing the state in the client (tearing down the connection). I feel that destroying the client (modifying the client state) should be done explicitly at the callsite with an subsequent `quit` or similar.

The semantics of this new API could also be changed to mean "invoke both client and server shutdown" but that conflates the responsibilities I think. For example, imagine the case where you deploy memcached behind a protocol-aware proxy (which we do); you might not necessarily want the client<>proxy connection destroyed after relaying the shutdown command between proxy<>server(s).

Reply:
Yes, I generally agree with you.

Because I'm not familiar with `shutdown`, could you say what happens from the client's perspective after a successful shutdown is issued? Is the client still usable (perhaps after a reconnect)?

Reply:
In general, the client should be usable after a reconnect. For the direct pymemcache <> server connection use case, I would say there likely isn't any practical reason to keep the local connection valid, but it's more a consideration for isolating the behavior of the interface (opting not to modify local state in a method designed to modify server state).

Another option could be adding a kwarg like `quit` or `close` whose default is `True` which optionally controls destroying the connection after successful shutdown. This could support use cases where one would want to manage the connection lifecycle manually/deploy behind a proxy/etc. while also supporting the more common use case.

Reply:
I think it's fine for clients who also want the `quit` behavior to call `quit()` explicitly after `shutdown()`.